### PR TITLE
[Mailer] [Amazon] Ensure "ext-simplexml" is loaded if SesApiTransport is used

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

For reference:
https://github.com/symfony/symfony/blob/150d8506e425212bbdc0e75b7d5fc4e6710f0852/src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesApiTransport.php#L66-L69

### CODE REVIEW

**Review by ogizanagi — APPROVED** [review]

**Review by jderusse — DISMISSED** [review]
Will you remember to remove this in 6.0?

https://github.com/symfony/symfony/blob/b47b239fd94a2f9016cc3dac306477ca7db698cb/src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesApiTransport.php#L24

**Comment by stof** [comment]
this should not become a hard requirement, as only one of the transports of the bridge need it

**Comment by OskarStark** [comment]
I agree that making it a hard requirement is bad for one who only uses one of the other transports, maybe checking for the existence of the class first and a clear exception message is a better approach.

@jderusse if this is merged we should definitely add an issue about that and set the 6.0 milestone.

Anyway it worked like this before but is not really clear and that's a big feeling I have when working with bridges of mailer and/or notifier. My main goal is to make those bridges more "stable" from a usage perspective, they do not feel very "Symfony". 

I am ok with closing this PR 👌🏻

**Comment by jderusse** [comment]
adding a check on the extension in `SesTransportFactory` sounds good to me.

**Comment by OskarStark** [comment]
> adding a check on the extension in SesTransportFactory sounds good to me.

Done @jderusse

File: src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php [hunk 1]
```diff
@@ -31,6 +32,10 @@ public function create(Dsn $dsn): TransportInterface
         $port = $dsn->getPort();
 
         if ('ses+api' === $scheme) {
+            if (!class_exists(\SimpleXMLElement::class)) {
+                throw new LogicException(sprintf('Cannot use "%s". Make sure you have "ext-simplexml" installed and enabled.', SesApiTransport::class));
```
Comment (jderusse) [inline src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php]: Beware when merging this in 5.1 @symfony/mergers : The message should only applied in `if (!class_exists(SesClient::class)) {` block

**Review by jderusse — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php [hunk 2]
```diff
@@ -31,6 +32,10 @@ public function create(Dsn $dsn): TransportInterface
         $port = $dsn->getPort();
 
         if ('ses+api' === $scheme) {
+            if (!class_exists(\SimpleXMLElement::class)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php]: Can't we replace this with an `\extension_loaded()` check? that'd allow opcache to optimize the code path.

Comment (OskarStark) [reply src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php]: Great idea 👍

Comment (OskarStark) [reply src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php]: Done

**Review by nicolas-grekas — APPROVED** [review]
(once fabbot is happy :) )

**Comment by OskarStark** [comment]
> (once fabbot is happy :) )

😄  done 👍

**Review by jderusse — APPROVED** [review]

**Comment by jderusse** [comment]
Thank you @OskarStark.

**Pull request closed without merging**